scripts/icons/utils.js

```diff
@@ -0,0 +1,41 @@
+// Files on which the rule doesn't apply
+const EXCEPTIONAL_ICONS = ["ConnectedDoc.svg"];
+const ALLOWED_COLORS = ["currentColor", "none"];
+
+function isEligibleForValidation(svgFile) {
+  return svgFile.endsWith(".svg") && !EXCEPTIONAL_ICONS.includes(svgFile);
+}
+
+function validateColors(colors, svgFileName, allowedColors = ALLOWED_COLORS) {
+  colors.forEach(({ attribute, color }) => {
+    if (!allowedColors.includes(color)) {
+      throw Error(generateErrorMessage(attribute, color, svgFileName));
+    }
+  });
+}
+
+function extractColorsFromSvg(content) {
+  const colorMatches = content.match(/(fill|stroke)="([\w#]+)"/g) || [];
```

Comment:
Let's replace here `"` with `["']` to be more precise

Reply:
Nice, correct, thanks!